### ISSUE TITLE
Columns removed in setup

### ISSUE THREAD

**Issue by tran-quang-thanh** [pycaret/pycaret#1131]
**Is your feature request related to a problem? Please describe.**
When I run setup and keep the default parameters, my X_train is transformed and it has fewer columns than the original data. I can keep the original columns by setting preprocess=False. What does preprocess do in this case? I don't know if setting preprocess=False will change anything else besides allowing all columns. Bellow is my logs when using default parameters.

**Additional context**
2021-03-25 10:30:04,048:INFO:Checking environment
2021-03-25 10:30:04,048:INFO:python_version: 3.8.7
2021-03-25 10:30:04,048:INFO:python_build: ('tags/v3.8.7:6503f05', 'Dec 21 2020 17:59:51')
2021-03-25 10:30:04,049:INFO:machine: AMD64
2021-03-25 10:30:04,049:INFO:platform: Windows-10-10.0.19041-SP0
2021-03-25 10:30:04,049:INFO:Memory: svmem(total=8421916672, available=2600263680, percent=69.1, used=5821652992, free=2600263680)
2021-03-25 10:30:04,049:INFO:Physical Core: 2
2021-03-25 10:30:04,049:INFO:Logical Core: 4
2021-03-25 10:30:04,049:INFO:Checking libraries
2021-03-25 10:30:04,050:INFO:pd==1.2.3
2021-03-25 10:30:04,050:INFO:numpy==1.19.5
2021-03-25 10:30:04,050:INFO:sklearn==0.23.2
2021-03-25 10:30:04,050:INFO:lightgbm==3.1.1
2021-03-25 10:30:04,050:INFO:catboost==0.24.4
2021-03-25 10:30:04,050:INFO:xgboost==1.3.3
2021-03-25 10:30:04,050:INFO:mlflow==1.14.1
2021-03-25 10:30:04,050:INFO:Checking Exceptions
2021-03-25 10:30:04,050:INFO:Declaring global variables
2021-03-25 10:30:04,051:INFO:USI: 4361
2021-03-25 10:30:04,051:INFO:pycaret_globals: {'seed', 'fold_shuffle_param', 'logging_param', '_all_models_internal', 'iterative_imputation_iters_param', 'transform_target_method_param', 'exp_name_log', 'experiment__', 'data_before_preprocess', 'transform_target_param', 'fold_generator', 'USI', 'fix_imbalance_method_param', 'target_param', 'y', '_all_metrics', '_internal_pipeline', 'master_model_container', 'imputation_regressor', '_available_plots', '_gpu_n_jobs_param', 'n_jobs_param', 'create_model_container', 'pycaret_globals', 'imputation_classifier', 'fold_param', 'y_train', 'stratify_param', 'X', 'prep_pipe', 'y_test', 'X_test', '_ml_usecase', 'fold_groups_param', 'X_train', 'fix_imbalance_param', '_all_models', 'display_container', 'gpu_param', 'html_param', 'log_plots_param'}
2021-03-25 10:30:04,051:INFO:Preparing display monitor
2021-03-25 10:30:04,051:INFO:Preparing display monitor
2021-03-25 10:30:04,072:INFO:Importing libraries
2021-03-25 10:30:04,073:INFO:Copying data for preprocessing
2021-03-25 10:30:04,096:INFO:Declaring preprocessing parameters
2021-03-25 10:30:04,122:INFO:Creating preprocessing pipeline
2021-03-25 10:30:04,395:INFO:Preprocessing pipeline created successfully
2021-03-25 10:30:04,396:ERROR:(Process Exit): setup has been interupted with user command 'quit'. setup must rerun.
2021-03-25 10:30:04,396:INFO:Creating global containers
2021-03-25 10:30:04,399:INFO:Internal pipeline: Pipeline(memory=None, steps=[('empty_step', 'passthrough')], verbose=False)
2021-03-25 10:30:07,839:INFO:Creating grid variables
2021-03-25 10:30:07,866:INFO:create_model_container: 0
2021-03-25 10:30:07,866:INFO:master_model_container: 0
2021-03-25 10:30:07,866:INFO:display_container: 1
2021-03-25 10:30:07,871:INFO:Pipeline(memory=None,
         steps=[('dtypes',
                 DataTypes_Auto_infer(categorical_features=[],
                                      display_types=True, features_todrop=[],
                                      id_columns=[], ml_usecase='regression',
                                      numerical_features=[],
                                      target='33550measure_icc_powerdown_vmin',
                                      time_features=[])),
                ('imputer',
                 Simple_Imputer(categorical_strategy='not_available',
                                fill_value_categorical=None,
                                fill_value_numeric...
                ('binn', 'passthrough'), ('rem_outliers', 'passthrough'),
                ('cluster_all', 'passthrough'),
                ('dummy', Dummify(target='33550measure_icc_powerdown_vmin')),
                ('fix_perfect',
                 Remove_100(target='33550measure_icc_powerdown_vmin')),
                ('clean_names', Clean_Colum_Names()),
                ('feature_select', 'passthrough'), ('fix_multi', 'passthrough'),
                ('dfs', 'passthrough'), ('pca', 'passthrough')],
         verbose=False)
2021-03-25 10:30:07,871:INFO:setup() succesfully completed......................................
2021-03-25 10:30:07,958:INFO:Initializing get_config()
2021-03-25 10:30:07,958:INFO:get_config(variable=X_train)

**Comment by ThinhNg** [comment]
I have the same issue, where it keeps dropping my columns as well.

For some reason, if I set data_split_stratify=True in the setup parameter, it fixes this issues.
e.g.

setupData = setup(data = data, target = 'Hit',normalize=True,normalize_method='minmax',data_split_stratify=True, fold=5,session_id=123)

This is using version pycaret 2.3.

I'm not sure why, but for some reason, the remove columns functionality is removing columns with us specifying.

**Comment by Yard1** [comment]
Columns that are perfectly correlated with another column are dropped by default. You can disable this with `remove_perfect_collinearity=False` in `setup()`.

**Comment by ThinhNg** [comment]
Got it. It worked after applying that setting!
Thanks @Yard1 .


What is strange though, the print out for the setup breakdown is still showing False regardless of whether I changed the setting to remove_perfect_collinearity=True.

![image](https://user-images.githubusercontent.com/14099198/112902173-626c3c80-90ab-11eb-91ee-777d18acd92e.png)

**Comment by Yard1** [comment]
That could be a small bug, I'll check. Thanks!

**Comment by tran-quang-thanh** [comment]
@ThinhNg I think remove_perfect_collinearity and remove_multicollinearity are different. In setup you can set those 2 separately, and by default, remove_multicollinearity=False but remove_perfect_collinearity=True. I think remove_perfect_collinearity is not shown in the breakdown

**Comment by ThinhNg** [comment]
@tran-quang-thanh Oh I see. Good catch. It would be nice to show it in the setup table.

**Comment by moezali1** [comment]
@tran-quang-thanh That's right. `remove_perfect_collinearity` and `remove_collinearity` are two different parameters. 

Also, `remove_perfect_collinearity` is not shown in the table, causing the confusion.

@tran-quang-thanh @ThinhNg Will you be able to push a commit to include `remove_perfect_collinearity` in the table. Should be an easy change.

@Yard1 Which branch should we use?

**Comment by Yard1** [comment]
@moezali1 `master`

**Comment by mjkanji** [comment]
@Yard1 @moezali1 I've submitted a PR that fixes this.
